src/atoms/atomgraph.jl

```diff
@@ -23,34 +24,42 @@ A type representing an atomic structure as a graph (`gr`).
   convolution operations by avoiding recomputing it every pass.
 - `id::String`: Optional, an identifier, e.g. to correspond with tags/labels of an imported
   dataset.
+- `structure::A`: the original representation from which this AtomGraph was created
 """
-mutable struct AtomGraph <: AbstractAtoms
+mutable struct AtomGraph{A} <: AbstractAtoms{A}
     graph::SimpleWeightedGraph{<:Integer,<:Real}
     elements::Vector{String}
     laplacian::Matrix{<:Real} # wanted to use LightGraphs.LinAlg.NormalizedGraphLaplacian but seems this doesn't support weighted graphs?
+    structure::A
     id::String # or maybe we let it be a number too?
 end
 
 # one without features or featurization initialized yet
 function AtomGraph(
-    graph::SimpleWeightedGraph{A,B},
+    graph::SimpleWeightedGraph{B,C},
     elements::Vector{String},
+    structure::A,
     id = "",
-) where {B<:Real,A<:Integer}
+) where {C<:Real,B<:Integer,A}
     # check that elements is the right length
     num_atoms = size(graph)[1]
     @assert length(elements) == num_atoms "Element list length doesn't match graph size!"
 
-    # this was previously B.(normalized_laplacian(graph)) - won't that potentially give rise to compatibility issues if B is a custom type?
+    # this was previously C.(normalized_laplacian(graph)) - won't that potentially give rise to compatibility issues if C is a custom type?
     laplacian = normalized_laplacian(graph)
-    AtomGraph(graph, elements, laplacian, id)
+    AtomGraph{A}(graph, elements, laplacian, structure, id)
```

Comment:
will the parameter get populated correctly otherwise? I thought it was clearer to be explicit since there's now a separate constructor for the case where `A==SimpleWeightedGraph`...

Reply:
yeah it should

Reply:
okay cool I'll take it out then and just leave a note to myself to make sure to write a test for that haha